presto-main/src/main/java/com/facebook/presto/sql/planner/TranslationMap.java

```diff
@@ -305,6 +305,6 @@ private Expression coerceIfNecessary(Expression original, Expression rewritten)
         return plan.getScope()
                 .tryResolveField(expression)
                 .filter(ResolvedField::isLocal)
-                .map(field -> plan.getFieldMappings().get(field.getHierarchyFieldIndex()));
+                .map(field -> requireNonNull(plan.getFieldMappings().get(field.getHierarchyFieldIndex())));
```

Comment:
This expression is already complex, maybe add this as an addition map step?

Reply:
Note this is `Optional#map`, not `Stream#map`. The intention here is to make sure we don't incidentally return `null` from mapping function turning something that should produce a result into `empty`.